### PR TITLE
Use redux in react components

### PR DESCRIPTION
**Pull request by flemton**
Dear reviewer, it is a pleasure to request a review for my PR before merging into dev
Below were the changes implemented;
- [Fix dependency cycle and add remove book events handler](https://github.com/flemton/bookstore/commit/fb31dfbea10556cbb07c377418ec26d8e7507e04)
- [Add event handler to add to books array](https://github.com/flemton/bookstore/commit/8c0707286dd3c820274073b713eca5a62d551b0a)
- [Add check status action](https://github.com/flemton/bookstore/commit/014819a87e899ef98697ea08d8ea38655cfe9236)

__Any changes requested are highly welcome, cheers 🥂__

### CODE REVIEW

**Review by victorokoroji — APPROVED** [review]
Hi @flemton 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Form.js [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
+        const author = e.target[1].value;
+        dispatch(addBook({ id: uuidv4(), title, author }));
+        e.target[0].value = '';
+        e.target[1].value = '';
+      })}
+      >
+        <input type="text" placeholder="Book title" required className="book-title" />
+        <input type="text" placeholder="Author" required className="book-author" />
+        <button className="addBook" type="submit">
+          ADD BOOK
```
Comment (roshan-bajgain) [inline src/components/Form.js:24]: - [ ] [Optional] I notice that your add and remove functionality is not working please fix this issue. when the user click the Add buttons then it should add the books. Please fix this issue

File: src/components/Book.js [hunk 2]
```diff
@@ -13,8 +26,9 @@ const Book = (props) => {
         <p className="author">{author}</p>
         <ul className="book-action">
           <li>Comments</li>
-          <li>Remove</li>
+          <button type="button" onClick={() => dispatch(removeBook({ id }))}>Remove</button>
           <li>Edit</li>
+          <button type="button" className="status-btn" onClick={() => dispatch(checkStatus({ id }))}>Status</button>
         </ul>
```
Comment (roshan-bajgain) [inline src/components/Book.js:32]: - [ ] [Optional] As I notice that remove is also not working please fix this issue. When the user click the remove buttons then It should remove the book with help of id.

**Review by roshan-bajgain — COMMENTED** [review]
Hi @flemton, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**